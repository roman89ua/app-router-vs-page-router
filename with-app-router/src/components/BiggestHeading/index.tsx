import { DetailedHTMLProps, HTMLAttributes } from "react";

function BiggestHeading({
  children,
  className,
  ...restProps
}: DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>) {
  return (
    <h1 className={`font-extrabold text-2xl ${className} `} {...restProps}>
      {children}
    </h1>
  );
}

BiggestHeading.defaultProps = {
  className: "",
};

export default BiggestHeading;
