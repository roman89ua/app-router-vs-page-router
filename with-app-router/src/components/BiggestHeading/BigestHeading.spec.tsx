import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import BiggestHeading from "@/components/BiggestHeading/index";

describe("BiggestHeading", () => {
  const headingText: string = "Heading text";

  it(`should render h1 with text "${headingText}"`, () => {
    render(<BiggestHeading>{headingText}</BiggestHeading>);
    const headingWithPredefinedText = screen.getByText(headingText);
    expect(headingWithPredefinedText).toBeInTheDocument();
    expect(headingWithPredefinedText).toMatchSnapshot();
  });

  it(`should render h1 with text "${headingText}" and proper color of the test`, () => {
    render(
      <BiggestHeading
        style={{
          color: "red",
        }}
      >
        {headingText}
      </BiggestHeading>
    );
    const headingWithPredefinedText = screen.getByText(headingText);
    const headingStyle = window.getComputedStyle(headingWithPredefinedText);
    expect(headingStyle.color).toBe("blue");
  });
  // 'text-red-600' ===  color: rgb(220 38 38) or #DC2626  according to tailwindcss text color https://tailwindcss.com/docs/text-color
});
