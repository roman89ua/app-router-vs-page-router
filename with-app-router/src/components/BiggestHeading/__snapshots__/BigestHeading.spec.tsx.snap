// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiggestHeading should render h1 with text "Heading text" 1`] = `
<h1
  class="font-extrabold text-2xl  "
>
  Heading text
</h1>
`;
